Add integration for exim (#1326)
DataDog/integrations-extras: exim/README.md

# Agent Check: exim

## Overview

This check monitors [exim][1] through the Datadog Agent.

## Setup

Follow the instructions below to install and configure this check for an Agent running on a host. For containerized environments, see the [Autodiscovery Integration Templates][3] for guidance on applying these instructions.

> ### Installation
> 
> The exim check is not included in the [Datadog Agent][2] package, so you need to install it.

Reviewer: ```suggestion
### Installation

For Agent v7.21+ / v6.21+, follow the instructions below to install the Exim check on your host. See [Use Community Integrations][10] to install with the Docker Agent or earlier versions of the Agent.

1. Run the following command to install the Agent integration:

   ```shell
   datadog-agent integration install -t datadog-exim==<INTEGRATION_VERSION>
   ```

2. Configure your integration similar to core [integrations][11].
```

2. [Restart the Agent][5].

### Validation

[Run the Agent's status subcommand][6] and look for `exim` under the Checks section.

## Data Collected

### Metrics

See [metadata.csv][7] for a list of metrics provided by this integration.

### Events

The exim integration does not include any events.

### Service Checks

See [service_checks.json][8] for a list of service checks provided by this integration.

## Troubleshooting

Need help? Contact [Datadog support][9].

[1]: https://www.exim.org/
[2]: https://app.datadoghq.com/account/settings#agent
[3]: https://docs.datadoghq.com/agent/kubernetes/integrations/
[4]: https://github.com/DataDog/integrations-core/blob/master/check/datadog_checks/check/data/conf.yaml.example
[5]: https://docs.datadoghq.com/agent/guide/agent-commands/#start-stop-and-restart-the-agent
[6]: https://docs.datadoghq.com/agent/guide/agent-commands/#agent-status-and-information
[7]: https://github.com/DataDog/integrations-core/blob/master/check/metadata.csv
[8]: https://github.com/DataDog/integrations-core/blob/master/check/assets/service_checks.json
[9]: https://docs.datadoghq.com/help/

> [10]: https://docs.datadoghq.com/developers/integrations/new_check_howto/#developer-toolkit

Reviewer: ```suggestion
[10]: https://docs.datadoghq.com/agent/guide/use-community-integrations/
[11]: https://docs.datadoghq.com/agent/guide/use-community-integrations/
```

# Agent Check: exim

## Overview

This check monitors [exim][1] through the Datadog Agent.

## Setup

Follow the instructions below to install and configure this check for an Agent running on a host. For containerized environments, see the [Autodiscovery Integration Templates][4] for guidance on applying these instructions.

### Installation

For Agent v7.21+ / v6.21+, follow the instructions below to install the exim check on your host. See [Use Community Integrations][3] to install with the Docker Agent or earlier versions of the Agent.

1. Run the following command to install the Agent integration:

   ```shell
   datadog-agent integration install -t datadog-exim==<INTEGRATION_VERSION>
   ```

> 2. Configure your integration similar to core [integrations][4].

Reviewer 2: ```suggestion
2. Configure your integration similar to core [integrations][2].
```

### Configuration

1. Edit the `exim.d/conf.yaml` file, in the `conf.d/` folder at the root of your Agent's configuration directory to start collecting your exim performance data. See the [sample exim.d/conf.yaml][5] for all available configuration options.

2. [Restart the Agent][6].

### Validation

[Run the Agent's status subcommand][7] and look for `exim` under the Checks section.

## Data Collected

### Metrics

See [metadata.csv][8] for a list of metrics provided by this integration.

### Events

The exim integration does not include any events.

### Service Checks

See [service_checks.json][9] for a list of service checks provided by this integration.

## Troubleshooting

Need help? Contact [Datadog support][10].

[1]: https://www.exim.org/

> [3]: https://app.datadoghq.com/account/settings#agent

Reviewer 2: ```suggestion
[2]: https://docs.datadoghq.com/getting_started/integrations/
[3]: https://docs.datadoghq.com/agent/guide/use-community-integrations/
```

# Agent Check: exim

## Overview

> This check monitors [exim][1] through the Datadog Agent.

Reviewer 2: ```suggestion
This check monitors [Exim][1] through the Datadog Agent.
```

## Setup

Follow the instructions below to install and configure this check for an Agent running on a host. For containerized environments, see the [Autodiscovery Integration Templates][4] for guidance on applying these instructions.

### Installation

For Agent v7.21+ / v6.21+, follow the instructions below to install the exim check on your host. See [Use Community Integrations][3] to install with the Docker Agent or earlier versions of the Agent.

1. Run the following command to install the Agent integration:

   ```shell
   datadog-agent integration install -t datadog-exim==<INTEGRATION_VERSION>
   ```

2. Configure your integration similar to core [integrations][2].

### Configuration

1. Edit the `exim.d/conf.yaml` file, in the `conf.d/` folder at the root of your Agent's configuration directory to start collecting your exim performance data. See the [sample exim.d/conf.yaml][5] for all available configuration options.

2. [Restart the Agent][6].

### Validation

[Run the Agent's status subcommand][7] and look for `exim` under the Checks section.

## Data Collected

### Metrics

See [metadata.csv][8] for a list of metrics provided by this integration.

### Events

> The exim integration does not include any events.

Reviewer 2: ```suggestion
The Exim integration does not include any events.
```

# Agent Check: exim

## Overview

This check monitors [exim][1] through the Datadog Agent.

## Setup

Follow the instructions below to install and configure this check for an Agent running on a host. For containerized environments, see the [Autodiscovery Integration Templates][3] for guidance on applying these instructions.

### Installation

To install the exim check on your host:

1. Install the [developer toolkit][10] on any machine.

2. Run `ddev release build exim` to build the package.

3. [Download the Datadog Agent][2].

4. Upload the build artifact to any host with an Agent and
 run `datadog-agent integration install -w

>  path/to/exim/dist/<ARTIFACT_NAME>.whl`.

Reviewer 3: Once this integration is released, there's no need to download the developer toolkit. Customers will just be able to install the integration with instructions [from here](https://docs.datadoghq.com/agent/guide/use-community-integrations/?tab=agentv721v621).

# Agent Check: exim

## Overview

This check monitors [exim][1] through the Datadog Agent.

## Setup

Follow the instructions below to install and configure this check for an Agent running on a host. For containerized environments, see the [Autodiscovery Integration Templates][3] for guidance on applying these instructions.

### Installation

To install the exim check on your host:

> 1. Install the [developer toolkit]
> (https://docs.datadoghq.com/developers/integrations/new_check_howto/#developer-toolkit)
>  on any machine.

Reviewer 2: ```suggestion
1. Install the [developer toolkit][10] on any machine.
```
